debug_module.tex

```diff
@@ -241,11 +243,11 @@ \section{Program Buffer} \label{programbuffer}
 While these programs are executed, the hart does not leave Debug Mode (see
 Section~\ref{debugmode}).  If an exception is encountered during execution of
 the Program Buffer, no more instructions are executed, the hart remains in Debug
-Mode, and \Fcmderr is set to 3({\tt exception error}).
+Mode, and \Fcmderr is set to 3 ({\tt exception error}).
 
 Executing the Program Buffer may clobber \Rdpc. If that is the case, it must be
 possible to read/write \Rdpc using an abstract command with \Fpostexec not set.
-The debugger must attempt to save \Rdpc between halting and
+The debugger must always save \Rdpc between halting and
```

Comment:
There were a lot of discussion about this. How is debugger supposed to save dpc before executing program buffer if the only way to access DPC is through program buffer? So, that is why it should be "attempt to". 

And, why do we need to do this at all?